packages/react-composites/src/composites/common/SingletonProvider.tsx

```diff
@@ -52,16 +52,23 @@ export interface BaseCompositeProps<TIcons extends Record<string, JSX.Element>>
 }
 
 /**
- * A base class for composites.
- * Provides common wrappers such as FluentThemeProvider and LocalizationProvider.
+ * A single {@link React.Context} to wrap components with the required providers (locale, icons, theme).
+ *
+ *  has ability check whether it is wrapped in another Singleton provider so that nested instances know not to layer providers
+ *  causing warnings and errors.
  *
  * @private
  */
-export const BaseComposite = (
+export const SingletonProvider = (
   props: BaseCompositeProps<CallCompositeIcons | ChatCompositeIcons> & { children: React.ReactNode }
 ): JSX.Element => {
   const { fluentTheme, rtl, locale } = props;
 
+  const alreadyWrapped = useSingleton();
```

Comment:
Curious if something like this would work where instead of having to have a singleton that checks if the composite has been wrapped we instead check if the icons themselves have already been registered:
```typescript
  // Only register icons that have not already been registered
  const icons = Object.entries({ ...DEFAULT_COMPOSITE_ICONS, ...props.icons });
  const iconsToBeRegistered = icons.filter(([iconName]) => IS_ALREADY_REGISTERED(iconName));
  registerIcons({ icons: Object.fromEntries(iconsToBeRegistered) });
```

There doesn't appear to be an immediate way from fluent's api to check if an Icon is already registered, looking at how they store their registered icons (https://github.com/microsoft/fluentui/blob/f26fbc8c9b7978efb5b3182ae586b74a5c93db77/packages/style-utilities/src/utilities/icons.ts#L60) it would be something like
```typescript
import { GlobalSettings, IIconRecord } from '@fluentui/react';
...
const previouslyRegisteredIcons = Object.keys(GlobalSettings.getValue<IIconRecords>('icons'));
const IS_ALREADY_REGISTERED = (iconName) => previouslyRegisteredIcon.includes(iconName.toLowerCase());
```
But we wouldn't want to do this unless a function exposed from fluent's library did this (we could contribute to the fluent library and add this function however)

Reply:
nuke option: 
![image](https://user-images.githubusercontent.com/82062616/149843905-1a6607e3-b6d0-47a3-b9a0-4501fb16bd85.png)


https://github.com/microsoft/fluentui/wiki/Using-icons#library-icon-registration

Reply:
Another options to solve this for meetings composite, is for the BaseComposite to take a prop called
```ts
interface BaseCompositeProps {
...
  registerIcons: boolean // default true
}
```
that meeting compsite can do
```tsx
<CallComposite registerIcons={false} />
...
<ChatComposite registerIcons={false} />
```